src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/client/BaseHMPPSClient.kt

```diff
@@ -0,0 +1,116 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.client
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpStatus
+import org.springframework.http.RequestEntity
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.context.request.RequestContextHolder
+import org.springframework.web.context.request.ServletRequestAttributes
+import java.net.URI
+
+abstract class BaseHMPPSClient(
```

Comment:
Might be worth thinking why this is better than using different webClient configurations and injecting those into the service.

Most services would use the clientRegistrationId and spring config to control this for instance https://github.com/ministryofjustice/hmpps-prisoner-from-nomis-migration/blob/5eb6af449b1a9a9551f31e57db0c5e5490f03989/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/config/WebClientConfiguration.kt

Reply:
Have switched over for consistency sake